tests/test_edgeql_functions.py

```diff
@@ -996,6 +996,309 @@ async def test_edgeql_functions_array_get_08(self):
+            'array index -5 is out of bounds'
+        ):
+            await self.con.query(
+                r'''SELECT array_insert([1, 2, 3, 4], -5, 9);''',
+            )
+
+    async def test_edgeql_functions_array_insert_04(self):
+        with self.assertRaisesRegex(
+            edgedb.InvalidValueError,
+            'array index 2 is out of bounds'
+        ):
+            await self.con.query(
+                r'''SELECT array_insert([1], 2, 9);''',
+            )
+
+    async def test_edgeql_functions_array_insert_05(self):
+        with self.assertRaisesRegex(
+            edgedb.InvalidValueError,
+            'array index -2 is out of bounds'
+        ):
+            await self.con.query(
+                r'''SELECT array_insert([1], -2, 9);''',
+            )
+
+    async def test_edgeql_functions_array_insert_06(self):
+        with self.assertRaisesRegex(
+            edgedb.InvalidValueError,
+            'array index 1 is out of bounds'
+        ):
+            await self.con.query(
+                r'''SELECT array_insert(<array<int64>>[], 1, 9);''',
+            )
+
+    async def test_edgeql_functions_array_insert_07(self):
+        with self.assertRaisesRegex(
+            edgedb.InvalidValueError,
+            'array index -1 is out of bounds'
+        ):
+            await self.con.query(
+                r'''SELECT array_insert(<array<int64>>[], -1, 9);''',
+            )
+
+    async def test_edgeql_functions_array_remove_01(self):
+        # Positive indexes
+        await self.assert_query_result(
+            r'''SELECT array_remove([1, 2, 3, 4], 0);''',
+            [[2, 3, 4]],
+        )
+
+        await self.assert_query_result(
+            r'''SELECT array_remove([1, 2, 3, 4], 1);''',
+            [[1, 3, 4]],
+        )
+
+        await self.assert_query_result(
+            r'''SELECT array_remove([1, 2, 3, 4], 2);''',
+            [[1, 2, 4]],
+        )
+
+        await self.assert_query_result(
+            r'''SELECT array_remove([1, 2, 3, 4], 3);''',
+            [[1, 2, 3]],
+        )
+
+        # Negative indexes
+        await self.assert_query_result(
+            r'''SELECT array_remove([1, 2, 3, 4], -1);''',
+            [[1, 2, 3]],
+        )
+
+        await self.assert_query_result(
+            r'''SELECT array_remove([1, 2, 3, 4], -2);''',
+            [[1, 2, 4]],
+        )
+
+        await self.assert_query_result(
+            r'''SELECT array_remove([1, 2, 3, 4], -3);''',
+            [[1, 3, 4]],
+        )
+
+        # Size 1 array
+        await self.assert_query_result(
+            r'''SELECT array_remove([1], 0);''',
+            [[]],
+        )
+
+    async def test_edgeql_functions_array_remove_02(self):
+        with self.assertRaisesRegex(
+            edgedb.InvalidValueError,
+            'array index 4 is out of bounds'
+        ):
+            await self.con.query(
+                r'''SELECT array_remove([1, 2, 3, 4], 4);''',
+            )
```

Comment:
The precedent of `array_get` is that you never get an out-of-bounds error, but instead you get `{}` (which is otherwise impossible) or whatever the default value that you specified.

This logic _should_ work for `array_remove` as well. Removing a non-existing element from an array is a NOP and the original array is unchanged. Instead I see "array index <blah> is out of bounds".